app/assets/javascripts/analytics.js.erb

```diff
@@ -2,6 +2,14 @@
 var element = document.querySelector('[data-module="cookie-banner"]')
 new GOVUK.Modules.CookieBanner().start($(element))
 
+var stripPII = function() {
+  const params = new URLSearchParams(window.location.search)
+  for (const [key, value] of params.entries()) {
+    params.set(key, "XXXX")
+  }
+  return location.protocol + '//' + location.host + location.pathname + params
```

Comment:
```suggestion
  return location.protocol + '//' + location.host + location.pathname + '?' + params
```

Without the '?' the url returned looks like: 
```
"http://localhost:5000/confirmationcontact_gp=XXXX&reference_number=XXXX"
```

I'm not sure if it's ok to add a trailing `?` to urls if there aren't params. It might be worth returning `location.protocol + '//' + location.host + location.pathname` if params is `null` before doing the loop?

Reply:
ahh! have amended :)